content/posts/2021/reana-client-python-api.md

```diff
@@ -0,0 +1,342 @@
+  'workflow_id': '6cd613eb-f2fb-411b-9601-c89599925759',
+  'workflow_name': 'root6-roofit'
+}
+```
+
+### Checking workflow status
+
+After starting our workflow we might want to check its current status.
+To inspect the workflow status using REANA client we could use [status](https://reana-client.readthedocs.io/en/maint-0.7/#reana-client-status)
+command. Similarly, it can be achieved with `get_workflow_status` function.
+Since it will take couple of seconds for all the computations to be finished
+while executing our workflow, we will use a loop to wait and regularly check
+what is the current status of our workflow:
+
+```python
+import time
+from reana_client.api.client import get_workflow_status
+
+while True:
+    status_details = get_workflow_status(my_workflow_name, my_reana_token)
+    print('Current status: ', status_details['status'])
+    if status_details['status'] == 'finished':
+        break
+    time.sleep(1)
+```
+
+Output:
+
+```python
+Current status:  running
+...
+Current status:  finished
+```
+
+### Checking workflow logs
+
+If our workflow has failed or in case we want to debug or investigate how our
+workflow works, we may want to display the logs of our workflow. To do that
+using REANA client we could use [logs](https://reana-client.readthedocs.io/en/maint-0.7/#reana-client-logs)
+command. In our case the logs can be fetched with `get_workflow_logs` function
+where we only need to specify workflow name and access token. Optionally, we
+can add such parameters as a list of `steps` whose logs we are interested in
+or `page` and `size` parameters to paginate the logs output if it gets too
+verbose. For example, let's say that we want to quickly check the last 15 lines
+of logs of the second `fitdata` step of our workflow. We can do that as follows:
+
+```python
+import json
+from reana_client.api.client import get_workflow_logs
+
+workflow_logs = get_workflow_logs(my_workflow_name, my_reana_token, ['fitdata'])
+job_logs = json.loads(workflow_logs['logs'])['job_logs']
+fitdata_logs = job_logs[next(iter(job_logs))]['logs']
+
+for item in fitdata_logs.split('\n')[-15:]:
+    print(item)
+```
+
+Output:
+
+```python
+ERR MATRIX APPROXIMATE
+ PARAMETER  CORRELATION COEFFICIENTS
+       NO.  GLOBAL      1      2      3      4      5
+        1  0.00000   1.000  0.000  0.000  0.000  0.000
+        2  0.00000   0.000  1.000  0.000  0.000  0.000
+        3  0.00000   0.000  0.000  1.000  0.000  0.000
+        4  0.00000   0.000  0.000  0.000  1.000  0.000
+        5  0.00000   0.000  0.000  0.000  0.000  1.000
+ ERR MATRIX APPROXIMATE
+[#1] INFO:Minization -- RooMinimizer::optimizeConst: deactivating const optimization
+[#1] INFO:Plotting -- RooAbsPdf::plotOn(model) directly selected PDF components: (bkg)
+[#1] INFO:Plotting -- RooAbsPdf::plotOn(model) indirectly selected PDF components: ()
+Info in <TCanvas::Print>: png file results/plot.png has been created
+```
+
+### Listing workspace files
+
```

Comment:
While the workflow is running or after it finished, we would usually want to list the output files which our workflow produced in its workspace. In command-line interface, we would use the `ls` CLI command. In Python, the corresponding function is `list_files()` function that operates also on the workflow name.

...


Example output: